Ethena | Ethereum (#215)
deficollective/defiscan: src/content/protocols/ethena/usde.md

stage_requirements:
  [
    [
      {
        text: "Assets are not in custody by a centralized entity",
        status: "unfixed",
      },
      { text: "All contracts are verified", status: "fixed" },
      { text: "Source-available codebase", status: "fixed" },
      { text: "Public documentation exists", status: "fixed" },
    ],
    [
      {
        text: "Upgrades with potential of “loss of funds” not protected with Exit Window >= 7 days OR a sufficient Security Council",
        status: "unfixed",
      },
      {
        text: "Dependency with a High centralization score without mitigation",
        status: "unfixed",
      },
      {
        text: "Frontend backups or self-hosting option exists",
        status: "fixed",
      },
    ],
    [
      {
        text: "Upgrades with potential of “loss of funds or unclaimed yield” not protected with onchain governance AND Exit Window >= 30 days",
        status: "unfixed",
      },
      {
        text: "Dependencies with High or Medium centralization score and no mitigations.",
        status: "unfixed",
      },
      { text: "Alternative third-party frontends exist", status: "fixed" },
    ],
  ]
---

# Summary

> Ethena is a synthetic dollar protocol built on Ethereum. Ethena's synthetic dollar, `USDe`, is fully backed by crypto assets (Bitcoin, Ethereum, Solana), yield from staked ETH through LSTs and funding rates earnings from short futures positions on CEXs. `USDe` is minted and redeemed by whitelisted KYCed users. Any user can acquire `USDe` on the open market and stake it to get access to yield sourced from the strategies and passed onto the users of the `USDe` stablecoin. `ENA` the governance token is freely tradeable and can be used to signal the direction of the protocol through offchain governance on [Snapshot](https://snapshot.box/#/s:ethenagovernance.eth).

Reviewer: ```suggestion
Ethena is a synthetic dollar protocol built on Ethereum. Ethena's synthetic dollar, `USDe`, is fully backed by crypto assets (Bitcoin, Ethereum, Solana), yield from staked ETH through Liquid Staking Tokens (LSTs) and funding rates earnings from short futures positions on CEXs. `USDe` is minted and redeemed by whitelisted KYCed users. Any user can acquire `USDe` on the open market and stake it to get access to yield sourced from the strategies and passed onto the users of the `USDe` stablecoin. `ENA` the governance token is freely tradeable and can be used to signal the direction of the protocol through offchain governance on [Snapshot](https://snapshot.box/#/s:ethenagovernance.eth).
```

![Architecture](../diagrams/Ethena.png)

## USDe Token

The owner of the `USDe` contract is the [Dev Multisig](#security-council). The [Dev Multisig](#security-council) can set any minter address, which can increase the `USDe` supply and mint `USDe` tokens to any address. The multisig is trusted to not abuse this power. The minter address is set to the `EthenaMinting` contract, which requires a transfer of accepted collateral, before minting `USDe`.

## Minting and Redeeming USDe

The Ethena stablecoin protocol allows whitelisted users to mint and redeem `USDe` through a flow that combines offchain quoting and onchain order submission with final execution by Ethena servers using EOAs with roles `MINTER_ROLE` and `REDEEMER_ROLE` (see [table](#role-permission-inside-ethenaminting)).

The process begins offchain, where a whitelisted user interacts with an Ethena-hosted pricing server to request a quote. This quote reflects the amount of collateral to be deposited and the expected `USDe` to be received. Using this information, the user constructs and signs an EIP-712 order. The signed order includes the benefactor address, collateral asset and amount of `USDe` minted.

> A central server operated by the Ethena team validates this order. It ensures the collateral is eligible, that the order does not destabilize the system, and that the minting or redemption action complies with both global and per-asset block-level caps. These caps are enforced to prevent drastic changes in `USDe` supply and drastic collateral exposures, which could also destabilize the system. Since the checks are on central servers, these checks do not have to be strictly enforced due to the centralized nature of compute and control, in contrast if this checks were enforced through smart contracts, they can be strictly enforced.

Reviewer: ### 
```suggestion
A central server operated by the Ethena team validates this order. It ensures the collateral is eligible, that the order does not destabilize the system, and that the minting or redemption action complies with both global and per-asset block-level caps. These caps are enforced to prevent drastic changes in `USDe` supply and drastic collateral exposures, which could also destabilize the system. Since the checks are on central servers, these checks do not have to be strictly enforced due to the centralized nature of compute and control. In contrast, they could be striclty enforced if they were happening onchain using smart contracts.
```

Users can be removed from the whitelist for minting and redeeming. If users ,that already minted `USDe`, are removed from the whitelist, they have to sell their `USDe` on the open market since they have no access to their deposited collateral which could result in the _loss of funds_ if the `USDe` price drops below the collateral value.

The `USDe` staking contract which allows users to stake `USDe` includes a blacklist feature. If a _Staker_ is blacklisted, they are prevented from staking or unstaking `USDe`. This could result in _loss of funds_ if users are unable to retrieve their staked `USDe` and and if they have to sell `sUSDe` on the open market at a discount. Blacklisting can lead to a complete _loss of funds_ for a blacklisted user if their `sUSDe` is burned from their wallet and redistributed to another address or equally to all _Stakers_. Blacklisting and seizing staker funds are two steps that are carried out by two permission owners.

> > Upgradeability score: High

Reviewer: Here I think you should also mention how strategies are assigned and how can they be changed?

> Autonomy score: High

## Exit Window

The actions of changing the minter role, blacklisting `USDe` stakers and seizing their funds are not protected by an _Exit Window_ and can be executed immediately, thus could harm the user immediately.

> Exit Window score: High

## Accessibility

The user interactions with the protocol can be divided into two parts: the minting and redeeming of `USDe` and the staking of `USDe`.

Users can host a frontend to mint and redeem `USDe` from https://github.com/ethena-labs/ethena-minting-client. For this action the _Accessibility_ risk is _Medium_.

For staking `USDe` the protocol does not provide a frontend to self host. The staking is only possible through the Ethena frontend. For this action the _Accessibility_ risk is _High_.

> Accessibility score: High

## Conclusion

> The USDe stablecoin protocol achieves High centralization risk scores for its _Upgradeability_, _Autonomy_, _Exit Window_ and _Accessibility_ dimensions. This would mean that the protocol ranks Stage 0, however the funds are stored in centralzed custody and thus is classified as _Others_.

Reviewer: ```suggestion
The USDe stablecoin protocol achieves _High_ centralization risk scores for its _Upgradeability_, _Autonomy_, _Exit Window_ and _Accessibility_ dimensions. This would mean that the protocol ranks Stage 0, however the funds are stored in centralzed custody and thus is classified as **Others**.
```

# Reviewer's Notes

No additional notes.

# Protocol Analysis

![Architecture](../diagrams/Ethena.png)

## USDe Token

The owner of the `USDe` contract is the [Dev Multisig](#security-council). The [Dev Multisig](#security-council) can set any minter address, which can increase the `USDe` supply and mint `USDe` tokens to any address. The multisig is trusted to not abuse this power. The minter address is set to the `EthenaMinting` contract, which requires a transfer of accepted collateral, before minting `USDe`.

## Minting and Redeeming USDe

> The Ethena stablecoin protocol allows whitelisted users to mint and redeem `USDe` through a flow that combines off-chain quoting and on-chain order submission with final execution by Ethena servers using EOAs with roles `MINTER_ROLE` and `REDEEMER_ROLE` (see [table](#role-permission-inside-ethenaminting)).

Reviewer: ```suggestion
The Ethena stablecoin protocol allows whitelisted users to mint and redeem `USDe` through a flow that combines offchain quoting and onchain order submission with final execution by Ethena servers using EOAs with roles `MINTER_ROLE` and `REDEEMER_ROLE` (see [table](#role-permission-inside-ethenaminting)).
```

> The process begins off-chain, where a whitelisted user interacts with an Ethena-hosted pricing server to request a quote. This quote reflects the amount of collateral to be deposited and the expected `USDe` to be received. Using this information, the user constructs and signs an EIP-712 order. The signed order includes the benefactor address, collateral asset and amount of `USDe` minted.

Reviewer: ```suggestion
The process begins offchain, where a whitelisted user interacts with an Ethena-hosted pricing server to request a quote. This quote reflects the amount of collateral to be deposited and the expected `USDe` to be received. Using this information, the user constructs and signs an EIP-712 order. The signed order includes the benefactor address, collateral asset and amount of `USDe` minted.
```

> A central server operated by the Ethena team validates this order. It ensures the collateral is eligible, that the order does not destabilize the system, and that the minting or redemption action complies with both global and per-asset block-level caps. These caps are enforced to prevent drastic changes in `USDe` supply and drastic collateral exposures, which could also destabilize the system.

Reviewer: I would emphasize that those are not strict "enforcements", since this is not onchain

> If all internal checks pass, the Ethena server submits the signed order on-chain by calling the `mint` or `redeem` function on the `EthenaMinting` contract. The EOAs used by the server must hold the `MINTER_ROLE` to call `mint`, or the `REDEEMER_ROLE` to call `redeem` (see [table](#role-permission-inside-ethenaminting)). On `mint`, the collateral is transferred from the whitelisted user to the `EthenaMinting` contract and `USDe` is minted into the users address by calling `mint` on the `USDe` contract. The `EthenaMinting` contract is assigned as the minter in `USDe` and is therefore the sole entity able to issue new tokens. On `redeem`, `USDe` is burned and the corresponding collateral is released to the user. No cooldown period exists for redeeming.

Reviewer: ```suggestion
If all internal checks pass, the Ethena server submits the signed order onchain by calling the `mint` or `redeem` function on the `EthenaMinting` contract. The EOAs used by the server must hold the `MINTER_ROLE` to call `mint`, or the `REDEEMER_ROLE` to call `redeem` (see [table](#role-permission-inside-ethenaminting)). On `mint`, the collateral is transferred from the whitelisted user to the `EthenaMinting` contract and `USDe` is minted into the users address by calling `mint` on the `USDe` contract. The `EthenaMinting` contract is assigned as the minter in `USDe` and is therefore the sole entity able to issue new tokens. On `redeem`, `USDe` is burned and the corresponding collateral is released to the user. No cooldown period exists for redeeming.
```

Multiple administrative permissions exist to modify this flow. The `DEFAULT_ADMIN_ROLE` in `EthenaMinting` can set or change the global mint and redeem caps using `setGlobalMaxMintPerBlock` and `setGlobalMaxRedeemPerBlock`. Furthermore, the `GATEKEEPER_ROLE` may call `disableMintRedeem`, which pauses all minting and redeeming operations. This function is designed to temporary preventing further minting or redeeming in the case of a vulnerability or instability. Users that have collateral deposited and are prohibited to access their collateral can lose funds if they have to sell it in the open market if the `USDe` trades below the collateral value. Restoring minting and redeeming functionality requires setting non-zero limits via the respective `setGlobalMaxMintPerBlock` and `setGlobalMaxRedeemPerBlock` functions which only can be executed by the `DEFAULT_ADMIN_ROLE`.

The `GATEKEEPER_ROLE` may also revoke `MINTER_ROLE` and `REDEEMER_ROLE` from addresses that currently own the respective roles by calling `removeMinterRole`, `removeRedeemerRole` or `removeCollateralManagerRole`. Which is a designed security measure in case of compromised addresses.

> Additionally, `removeWhitelistedBenefactor` allows the `GATEKEEPER_ROLE` to remove an address from the whitelist. Only whitelisted addresses are allowed to supply valid orders. Ethena will not accept orders from addresses that are not whitelisted, to comply with AML laws and KYC requirements, as well as anti-terrorist financing laws. If an address created a position with collateral, but got removed from the whitelist, the address has to sell the `USDe` on the open market and has no access to its deposited collateral. This could lead to _loss of funds_ for users that have collateral deposited and are removed from the whitelist if the `USDe` price drops below the collateral value.

Reviewer: So they can't freeze funds with this whitelist? 
many "compliant" stablecoins have an additional whitelists such that they can freeze funds and the accounts can no longer transfer the tokens

> The [Dev Multisig](#security-council) owns the `DEFAULT_ADMIN_ROLE` and can replace the `GATEKEEPER_ROLE` with another address and stays the ultimate on-chain authority over the `USDe` supply.

Reviewer: ```suggestion
The [Dev Multisig](#security-council) owns the `DEFAULT_ADMIN_ROLE` and can replace the `GATEKEEPER_ROLE` with another address and keeps onchain authority over the `USDe` supply.
```

Collateral flows to custodian addresses are executed through the `transferToCustody` function, callable only by addresses with the `COLLATERAL_MANAGER_ROLE`. The destination addresses for this function can only be added or removed by holding the `DEFAULT_ADMIN_ROLE`. Maliciously set addresses can lead to stealing collateral from users and the protocol.

> All core permissions for minting, redeeming, managing roles, and custodial flows are ultimately under the control of [Dev Multisig](#security-council), either directly or indirectly via its ability to control `DEFAULT_ADMIN_ROLE` holders.

Reviewer: I find those paragraphs quite redundant, for instance everything mentioned here is said multiple times above

Yield from the various strategies is transferred into the system via the `StakingRewardsDistributor` contract. The `StakingRewardsDistributor` contract holds `USDe` rewards and regularly calls `transferInRewards` to transfer them into `StakedUSDeV2`. The `StakingRewardsDistributor` is operated by an address labeled [Operator](#security-council), which is itself appointed by the [Dev Multisig](#security-council). While the [Operator](#security-council) can delay transfers, they cannot redirect funds or change the recipient. This minimizes the risk of _loss of unclaimed yield_, though misconfiguration or inaction may still delay yield distribution. The [Dev Multisig](#security-council) can replace the [Operator](#security-council) to prevent _loss of unclaimed yield_ at any time.

The ability to rescue tokens from the vault, handled by `rescueTokens`, is restricted to addresses with the `DEFAULT_ADMIN_ROLE`. Although this function cannot extract `USDe`, it can affect other tokens accidentally sent to the vault.

# Dependencies

> `USDe` is only slightly overcollateralized and relies on various strategies to maintain stability and peg to the US-Dollar. The strategies rely on various counterparties to function correctly. If any of the counterparties fail, the solvency of the protocol is at risk. Ethena has a [Reserve Fund](#security-council) to provide a buffer in case of negative performance of the diversified yield strategy. Ethena employs diversification within each strategy by relying on various counterparties (various CEXs, different Custodians, diverse centralized Stablecoins and LSTs), as well as diversify in different strategies (short futures, staking yield, long spot).

Reviewer: ```suggestion
`USDe` is only slightly overcollateralized and relies on various strategies to maintain stability and peg to the US-Dollar. The strategies rely on various counterparties to function correctly. If any of the counterparties fail, the solvency of the protocol is at risk. Ethena has a [Reserve Fund](#security-council) to provide a buffer in case of negative performance of the diversified yield strategies. Ethena employs diversification within each strategy by relying on various counterparties (various CEXs, different Custodians, diverse centralized Stablecoins and LSTs), as well as diversify in different strategies (short futures, staking yield, long spot).
```

> `USDe` is only slightly overcollateralized and relies on various strategies to maintain stability and peg to the US-Dollar. The strategies rely on various counterparties to function correctly. If any of the counterparties fail, the solvency of the protocol is at risk. Ethena has a [Reserve Fund](#security-council) to provide a buffer in case of negative performance of the diversified yield strategy. Ethena employs diversification within each strategy by relying on various counterparties (various CEXs, different Custodians, diverse centralized Stablecoins and LSTs), as well as diversify in different strategies (short futures, staking yield, long spot).

Reviewer: In the second paragraph, I would just say that you list those below instead of them again

OES providers that Ethena uses include Copper, Ceffu and Fireblocks.

## Centralized Stablecoins

Ethena allocates parts of their collateral funds into `USDC` to earn yield distributed by Coinbase. Ethena therefore relies on Circle as Stablecoin issuer and on Coinbase custody.

## Backing Assets

Ethena holds `BTC`, `ETH`, `SOL` and ETH `LSTs` as collateral for the `USDe` stablecoin. To reduce dependency risk on a single `LST`, Ethena diversifies their collateral assets into Binance Staked ETH (WBETH), Lido stETH and Mantle's mETH.

## Margin Collateral

Most of the margin in perpetual futures positions are denominated in `USDT`. By this, Ethena is unidirectionally LONG exposed to the `USDT` price without offsetting short exposure. In case of `USDT` solvency risk, custodial risk, regulatory risk, Ethena would aim to open inverse perpetuals contracts, margined with `BTC` or `ETH`.

## USDtb

> `USDtb` is an asset in the [Reserve Fund](#security-council) to provide a buffer in case of negative performance of the diversified yield strategy.

Reviewer: Can you explain what is the token and who issues it

# Governance

> Ethena governance using the `ENA` token is purely used for signalling and unlike on-chain governance contracts is not technically enforcing. Holders of `ENA` participate in off-chain voting via Snapshot, where they may submit and vote on proposals related to risk parameters, vault configurations, supported collateral, and other protocol adjustments.

Reviewer: ```suggestion
Ethena governance using the `ENA` token is purely used for signalling and is not technically enforcing. Holders of `ENA` participate in offchain voting via Snapshot, where they may submit and vote on proposals related to risk parameters, vault configurations, supported collateral, and other protocol adjustments.
```

> Votes cast on Snapshot are technically not binding. Their outcomes inform the Ethena Labs Team and the Risk Committee, who are responsible for executing changes on-chain through controlled permission owners like the [Dev Multisig](#security-council). The Risk Committee consists of selected individuals or entities elected by the Ethena Labs Team.

Reviewer: ```suggestion
Votes cast on Snapshot are technically not binding. Their outcomes inform the Ethena Labs Team and the Risk Committee, who are responsible for executing changes onchain through controlled permission owners like the [Dev Multisig](#security-council). The Risk Committee consists of selected individuals or entities elected by the Ethena Labs Team.
```

> The `ENA` token itself is governed by the [Dev Multisig](#security-council), which holds exclusive access to the mint function. Token minting is constrained to a maximum of 10% annually and can only occur once every year. The ownership of `ENA` is non-renounceable, but transferable via `transferOwnership`.

Reviewer: so, technically still renounceable

Ultimately _Governance_ execution remains under centralized control.

## Security Council

> New table with all the multisigs

Reviewer: ```suggestion
```
